src/main/java/betterquesting/api2/client/gui/controls/PanelButtonQuest.java

```diff
@@ -34,45 +34,47 @@
 {
     public final GuiRectangle rect;
     public final EntityPlayer player;
-    
-    public PanelButtonQuest(GuiRectangle rect, int id, String txt, DBEntry<IQuest> value)
+	public final IGuiTexture txFrame;
+
+	public PanelButtonQuest(GuiRectangle rect, int id, String txt, DBEntry<IQuest> value)
     {
-        super(rect, id, txt, value);
+		super(rect, id, txt, value);
+		IGuiTexture txFrame1 = null;
         this.rect = rect;
         
         player = Minecraft.getMinecraft().thePlayer;
         EnumQuestState qState = value == null ? EnumQuestState.LOCKED : value.getValue().getState(player);
-        IGuiTexture txFrame = null;
         IGuiColor txIconCol = null;
         boolean main = value == null ? false : value.getValue().getProperty(NativeProps.MAIN);
         boolean lock = false;
         
         switch(qState)
         {
             case LOCKED:
-                txFrame = main ? PresetTexture.QUEST_MAIN_0.getTexture() : PresetTexture.QUEST_NORM_0.getTexture();
+                txFrame1 = main ? PresetTexture.QUEST_MAIN_0.getTexture() : PresetTexture.QUEST_NORM_0.getTexture();
                 txIconCol = PresetColor.QUEST_ICON_LOCKED.getColor();
                 lock = true;
                 break;
             case UNLOCKED:
-                txFrame = main ? PresetTexture.QUEST_MAIN_1.getTexture() : PresetTexture.QUEST_NORM_1.getTexture();
+                txFrame1 = main ? PresetTexture.QUEST_MAIN_1.getTexture() : PresetTexture.QUEST_NORM_1.getTexture();
                 txIconCol = PresetColor.QUEST_ICON_UNLOCKED.getColor();
                 break;
             case UNCLAIMED:
-                txFrame = main ? PresetTexture.QUEST_MAIN_2.getTexture() : PresetTexture.QUEST_NORM_2.getTexture();
+                txFrame1 = main ? PresetTexture.QUEST_MAIN_2.getTexture() : PresetTexture.QUEST_NORM_2.getTexture();
                 txIconCol = PresetColor.QUEST_ICON_PENDING.getColor();
                 break;
             case COMPLETED:
-                txFrame = main ? PresetTexture.QUEST_MAIN_3.getTexture() : PresetTexture.QUEST_NORM_3.getTexture();
+                txFrame1 = main ? PresetTexture.QUEST_MAIN_3.getTexture() : PresetTexture.QUEST_NORM_3.getTexture();
                 txIconCol = PresetColor.QUEST_ICON_COMPLETE.getColor();
                 break;
 			case REPEATABLE:
-				txFrame = main ? PresetTexture.QUEST_MAIN_4.getTexture() : PresetTexture.QUEST_NORM_4.getTexture();
+				txFrame1 = main ? PresetTexture.QUEST_MAIN_4.getTexture() : PresetTexture.QUEST_NORM_4.getTexture();
 				txIconCol = PresetColor.QUEST_ICON_REPEATABLE.getColor();
 				break;
         }
-        
-        IGuiTexture btnTx = new GuiTextureColored(txFrame, txIconCol);
+
+		txFrame = txFrame1;
```

Comment:
why was this added?

Reply:
Because txFrame are marked as final. IDEA suggest me to do this

Reply:
I don't think you need to introduce the txFrame1 variable at all, tbh. You can leave this as it was before...